/*
team_library.proto: Protobuf representation of a team library for the
scoreboard.

Copyright 2023-2024 Tracy Beck

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

syntax = "proto3";
package cszb_scoreboard.proto;

import "config.proto";

message TeamLibInfo {
  string name = 1;
  TeamInfo.TeamType default_team_type = 2;
  string image_path = 3;
  // is_relative is unused at the moment, but is reserved for future use to
  // match the image library.
  bool is_relative = 4;
}

message TeamLibrary {
  repeated TeamLibInfo teams = 1;
  // Library root is unused at the moment, but is reserved for future use to
  // match the image library.
  string library_root = 2;
}

// Data structure to handle the return values from the TeamLibraryDialog
message TeamLibraryDialogResponse {
  TeamLibInfo home = 1;
  TeamLibInfo away = 2;
}
